### PR TITLE
PAYEE-341 - Wip - Caching update income value for 48 hours and compar…

### PR DESCRIPTION
**Pull request by brendanstraughan**
…ing to nps value

- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/viewModels/IncomeSourceSummaryViewModel.scala [hunk 1]
```diff
@@ -70,6 +72,11 @@ object IncomeSourceSummaryViewModel {
     val displayAddCompanyCar =
       !benefitVMs.map(_.name).contains(Messages("tai.taxFreeAmount.table.taxComponent.CarBenefit"))
 
+    val isUpdateInProgress = cacheUpdatedIncomeAmount match {
+      case Some(cacheUpdateAMount) => if (cacheUpdateAMount != taxCodeIncomeSource.amount.toInt) true else false
```
Comment (sergi-hmrc) [inline app/uk/gov/hmrc/tai/viewModels/IncomeSourceSummaryViewModel.scala:76]: just return cacheUpdateAMount != taxCodeIncomeSource.amount.toInt

**Review by vivekhmrc — APPROVED** [review]

File: app/controllers/IncomeController.scala [hunk 2]
```diff
@@ -309,32 +311,34 @@ class IncomeController @Inject()(
...
-            logger.warn(s"Unable to parse updateIncomeDateKey $exception")
-            None
-        }
+      journeyCacheService
+        .collectedJourneyValues(
+          Seq(UpdateIncome_PayToDateKey, UpdateIncome_IdKey, UpdateIncome_NameKey),
+          Seq(UpdateIncome_DateKey))
+        .getOrFail flatMap {
+        case (mandatorySeq, optionalSeq) =>
+          val date = Try(optionalSeq.head.map(date => LocalDate.parse(date))) match {
```
Comment (sergi-hmrc) [inline app/controllers/IncomeController.scala:324]: no need for a Try or anything here... (Throwing an exception is expensive and useless in this case :D ) 
optionalSeq.headOption.map(date => LocalDate.parse(date))

File: app/controllers/income/estimatedPay/update/IncomeUpdateEstimatedPayController.scala [hunk 3]
```diff
@@ -89,57 +89,61 @@ class IncomeUpdateEstimatedPayController @Inject()(
...
+              Ok(incorrectTaxableIncome(payYearToDate, paymentDate.getOrElse(LocalDate.now), incomeSource.id)))
+        }
       }
-    }
 
-    result.value.map(_.sequence).flatMap(identity).map {
-      case Some(result) => result
-      case None         => Redirect(controllers.routes.TaxAccountSummaryController.onPageLoad())
-    }
+      result.value.map(_.sequence).flatMap(identity).map {
```
Comment (sergi-hmrc) [inline app/controllers/income/estimatedPay/update/IncomeUpdateEstimatedPayController.scala:144]: identity is not needed, just do `result.value.flatMap(_.sequence)`

File: app/controllers/IncomeController.scala (hunk 2 above)
Comment (brendanstraughan) [reply app/controllers/IncomeController.scala:324]: This will make date an option so will have extract this out, how do you want me to do thi?

Comment (sergi-hmrc) [reply app/controllers/IncomeController.scala:324]: it's fine as it is

**Review by sergi-hmrc — APPROVED** [review]

**Review by vivekhmrc — APPROVED** [review]
LGTM

**Review by sergi-hmrc — APPROVED** [review]

**Review by vivekhmrc — APPROVED** [review]

**Pull request merged**